src/app/fdctl/run/tiles/fd_store_int.c

```diff
@@ -0,0 +1,604 @@
+    if( slot_meta == NULL ) {
+      FD_LOG_ERR(( "could not find slot meta" ));
+    }
+
+    fd_hash_t const * block_hash = fd_blockstore_block_hash_query( ctx->blockstore, slot );
+    if( block_hash == NULL ) {
+      FD_LOG_ERR(( "could not find slot meta" ));
+    }
+
+    FD_STORE( ulong, out_buf, slot_meta->parent_slot );
+    out_buf += sizeof(ulong);
+
+    memcpy( out_buf, block_hash->uc, sizeof(fd_hash_t) );
+    out_buf += sizeof(fd_hash_t); 
+
+    uchar * block_data = fd_blockstore_block_data_laddr( ctx->blockstore, block );
+
+    FD_SCRATCH_SCOPE_BEGIN {
+      fd_block_info_t block_info;
+      fd_runtime_block_prepare( block_data, block->data_sz, fd_scratch_virtual(), &block_info );
+    
+      FD_LOG_NOTICE(( "block prepared - slot: %lu", slot ));
+
+      fd_txn_p_t * txns = fd_type_pun( out_buf );
+      ulong txn_cnt = fd_runtime_block_collect_txns( &block_info, txns );
+
+      ulong out_sz = sizeof(ulong) + sizeof(fd_hash_t) + ( txn_cnt * sizeof(fd_txn_p_t) );
+      fd_mcache_publish( ctx->replay_out_mcache, ctx->replay_out_depth, ctx->replay_out_seq, replay_sig, ctx->replay_out_chunk, txn_cnt, 0UL, tsorig, tspub );
+      ctx->replay_out_seq   = fd_seq_inc( ctx->replay_out_seq, 1UL );
+      ctx->replay_out_chunk = fd_dcache_compact_next( ctx->replay_out_chunk, out_sz, ctx->replay_out_chunk0, ctx->replay_out_wmark );
+    } FD_SCRATCH_SCOPE_END;
+  }
+
+  if( repair_req_cnt != 0 ) {
+    ulong tspub = fd_frag_meta_ts_comp( fd_tickcount() );
+    ulong repair_req_sig = 50UL;
+    ulong repair_req_sz = repair_req_cnt;
+    fd_mcache_publish( ctx->repair_req_out_mcache, ctx->repair_req_out_depth, ctx->repair_req_out_seq, repair_req_sig, ctx->repair_req_out_chunk,
+      repair_req_sz, 0UL, tsorig, tspub );
+    ctx->repair_req_out_seq   = fd_seq_inc( ctx->repair_req_out_seq, 1UL );
+    ctx->repair_req_out_chunk = fd_dcache_compact_next( ctx->repair_req_out_chunk, repair_req_sz, ctx->repair_req_out_chunk0, ctx->repair_req_out_wmark );
+  }
+
+  return;
+}
+
+static void
+during_housekeeping( void * _ctx ) {
+  fd_store_tile_ctx_t * ctx = (fd_store_tile_ctx_t *)_ctx;
+
+  fd_mcache_seq_update( ctx->replay_out_sync, ctx->replay_out_seq );
+  fd_mcache_seq_update( ctx->repair_req_out_sync, ctx->repair_req_out_seq );
+
+  ctx->store->now = fd_log_wallclock();
+
+  for( ulong i = fd_pending_slots_iter_init( ctx->store->pending_slots );
+         (i = fd_pending_slots_iter_next( ctx->store->pending_slots, ctx->store->now, i )) != ULONG_MAX; ) {
+    uchar const * block = NULL;
+    ulong         block_sz = 0;
+    ulong repair_slot = FD_SLOT_NULL;
+    int store_slot_prepare_mode = fd_store_slot_prepare( ctx->store, i, &repair_slot, &block, &block_sz );
+
+    ulong slot = repair_slot == 0 ? i : repair_slot;
+    FD_LOG_DEBUG(( "store slot - mode: %d, slot: %lu, repair_slot: %lu", store_slot_prepare_mode, i, repair_slot ));
+    fd_store_tile_slot_prepare( ctx, store_slot_prepare_mode, slot );
```

Comment:
Pretty suspicious that this call belongs in housekeeping? I suspect you want this to be in after_credit and have your own timer to fire it periodically.